### PR TITLE
return book - implementation of functionality

### PR DESCRIPTION
**Pull request by BartoliniAndBorderCollies**
### Description:
This feature implements the crucial functionality for library - possibility to return a book

### Motivation:
Library app must have feature to return a book

### Changes introduced by this pull request:

- add command `return book`
- add communication with user to get to know which book he wants to return
- add logic to obtain a list of books of a specific account
- add method to remove a book from a list of borrowed books
- update of amount of available books
- add `equals()` to have a possibility to identify the same object and remove it from the `lending_info` table

### Testing

- [x] application provides a list of books of a specific account
- [x] `return book` removes a book from the table of borrowed books
- [x] after returning a book the amount of this book availability raises

### CODE REVIEW

File: src/main/java/org/klodnicki/service/BookService.java [hunk 1]
```diff
@@ -39,4 +42,25 @@ public void update(BookInfo bookInfo) {
...
+        List<String> results = new ArrayList<>();
+        for(int i = 0; i< foundBooks.size(); i++) {
+            results.add(foundBooks.get(i).toString());
+        }
+        return results;
+    }
+
+    public List<String> prepareListOfBooks(String title, String author) {
+        List<String> results = new ArrayList<>();
+        for (int i = 0; i < findBooksByTitleAndAuthor(title, author).size(); i++) {
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/service/BookService.java]: This is incorrect due to the doubling the select query in database. It is possible for this query to return different lists - that's the first issue - and it also results in weak performance.

File: src/main/java/org/klodnicki/service/BookService.java [hunk 2]
```diff
@@ -39,4 +42,25 @@ public void update(BookInfo bookInfo) {
     public List<BookInfo> findBooksByTitleAndAuthor(String title, String author) {
         return bookRepository.findBooksByTitleAndAuthor(title, author);
     }
+
+    public List<String> prepareListOfBorrowedBooksByAccount (Account account) {
```
Comment (JasinskiSz) [inline src/main/java/org/klodnicki/service/BookService.java]: This method and prepareListOfBooks(Account account) should be changed. This method is not serving purpose, you can just directly use method prepareListOfBooks if you make it public. The other problem is that prepareListOfBooks doesn't only prepare, but also search for the books in database which should be done by separate method.

**Review by JasinskiSz — CHANGES_REQUESTED** [review]

File: src/main/java/org/klodnicki/service/BookService.java (hunk 2 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/org/klodnicki/service/BookService.java]: Ok I understand. that way was very legible to me but I understand and I changed that. Please check now.

File: src/main/java/org/klodnicki/service/BookService.java (hunk 1 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/org/klodnicki/service/BookService.java]: understood! Changed and ready to be re-viewed!

**Review by JasinskiSz — APPROVED** [review]

**Pull request merged**